### PR TITLE
Resolver fixes

### PR DESCRIPTION
**Pull request by iulianpascalau**
- fixed the following resolver-requester issues:
    - fixed reporting of missing transactions and miniblocks in resolvers to the debugging component
    - replaced function that gets the data from persister in miniblock resolver from `Get` to `SearchFirst` as to align with the rest of the resolvers.
- removed unused flags in main.go
- fixed a wrong iteration on an int slice

Testing:
Launch a normal testnet with #AllIn, things should go pretty smooth and no node should be left behind especially at the epoch change event.

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: cmd/node/main.go [hunk 1]
```diff
@@ -353,20 +353,6 @@ VERSION:
 			"and will have a full history over epochs.",
 	}
 
-	numEpochsToSave = cli.Uint64Flag{
```
Comment (bogdan-rosianu) [inline cmd/node/main.go:356]: why did you remove the flags?

Comment (iulianpascalau) [reply cmd/node/main.go:356]: no longer required, they are defined in the config.toml and observed that we never used them as flags. Initially I thought we are mistakenly changed those when running the node. Didn't turned out that we specify them in the testnet scripts.

Comment (bogdan-rosianu) [reply cmd/node/main.go:356]: Ok, just wondering if one would have used these flags in a custom node script. It is easier to configure these settings via flags.

Comment (iulianpascalau) [reply cmd/node/main.go:356]: It is very easy to mess things out. I was even wondering if we can remove some configs values from config.toml (regarding this area) as to further decrease the chances of breaking things up.

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.
